bot/exts/info/python_news.py

```diff
@@ -75,8 +75,17 @@ async def get_webhook_names(self) -> None:
 
     @staticmethod
     def escape_markdown(content: str) -> str:
-        """Escape the markdown underlines and spoilers."""
-        return re.sub(r"[_|]", lambda match: "\\" + match[0], content)
+        """Escape the markdown underlines and spoilers that aren't in codeblocks."""
```

Comment:
Can we have explanations for this regex? Potentially also move it to `bot/utils/regex.py`.

Reply:
I've redesigned this PR quite a bit.

The regex does now have comments explaining each part, and also a comment above where it gets used to explain why the regex works.

In terms of moving it, I decided to instead just make it a constant within the file.